File: gamelogic/src/main/java/com/alta_v2/game/gamelogic/stage/MenuStage.java
package com.alta_v2.game.gamelogic.stage;

import com.alta_v2.game.dao.facade.definition.DefinitionDaoApi;
import com.alta_v2.game.gamelogic.stage.event.ChangeMenuStageEvent;
import com.google.inject.Inject;
import lombok.RequiredArgsConstructor;
import lombok.extern.log4j.Log4j2;

@Log4j2
@RequiredArgsConstructor(onConstructor = @__(@Inject))
public class MenuStage extends AbstractStage {

    private final DefinitionDaoApi definitionDaoApi;

    /**
     * {@inheritDoc}
     */
    @Override
    public void onActionBegin(ActionType action) {
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void onActionFinish(ActionType action) {
        if (action != ActionType.BACK) {
            return;
        }

        ChangeMenuStageEvent event = new ChangeMenuStageEvent(definitionDaoApi.getMapDefinition(1001));
        this.changeStage(event);
    }
}

File: gamelogic/src/main/kotlin/com/alta_v2/game/gamelogic/stage/MenuStage.kt
package com.alta_v2.game.gamelogic.stage

import com.alta_v2.game.gamelogic.domain.map.MapDaoService
import com.alta_v2.game.gamelogic.stage.event.ChangeMenuStageEvent
import com.alta_v2.mediator.serde.ActionType
import com.google.inject.Inject
import mu.KotlinLogging

class MenuStage @Inject constructor(private val mapDaoService: MapDaoService) : AbstractStage() {

    private val log = KotlinLogging.logger {  }

    override fun onActionBegin(action: ActionType) {}

    override fun onActionFinish(action: ActionType) {
        if (action != ActionType.BACK) {
            return
        }

        val mapDefinition = mapDaoService.getDefinition(1001)
        if (mapDefinition == null) {
            log.warn("Map definition for given mapId: 1001 is null")
            return
        }

        val event = ChangeMenuStageEvent(mapDefinition)
        changeStage(event)
    }
}